### PR TITLE
FIX-#2405: Make sure named aggregation work for Series objects

### LINKED ISSUE #2405: Named aggregation doesn't work with Series objects
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Ubuntu 20.04

- **Modin version** (`modin.__version__`):

0.8.2.1+6.g2b0b755

- **Python version**:

Python 3.7.8

- **Code we can use to reproduce**:

```python
import modin.pandas as pd

s1 = pd.Series([10, 10, 10, 1, 1, 1, 2, 3])
s1.name = "data"

print(s1)
print(type(s1))
gb = s1.groupby(level=0)
print(gb.groups)
s2 = gb.agg(result=("max"))
print(s2)
print(type(s2))
```

### Describe the problem

On Modin this code throws exception from Pandas `reconstruct_func` because it doesn't accept such arguments.
```
Traceback (most recent call last):
  File "groupby_test24.py", line 10, in <module>
    s2 = gb.agg(result=("max"))
  File "/localdisk/gashiman/modin/modin/pandas/groupby.py", line 376, in aggregate
    func, **kwargs
  File "/localdisk/gashiman/modin/modin/pandas/groupby.py", line 364, in _reconstruct_func
    func, **kwargs
  File "/localdisk/gashiman/miniconda3/envs/modin_on_omnisci/lib/python3.7/site-packages/pandas/core/aggregation.py", line 86, in reconstruct_func
    raise TypeError("Must provide 'func' or tuples of '(column, aggfunc).")
TypeError: Must provide 'func' or tuples of '(column, aggfunc).
```

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2405 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@dchigarev could you take a look?

File: modin/pandas/base.py [hunk 1]
```diff
@@ -655,6 +655,7 @@ def aggregate(self, func=None, axis=0, *args, **kwargs):  # noqa: PR01, RT01, D2
         Aggregate using one or more operations over the specified axis.
         """
         axis = self._get_axis_number(axis)
+
```
Comment (YarShev) [inline modin/pandas/base.py:658]:
```suggestion
Seems not related.
```

File: modin/pandas/groupby.py [hunk 2]
```diff
@@ -1924,8 +1924,45 @@ def nsmallest(self, n=5, keep="first"):
             )
         )
 
+    def _validate_func_kwargs(self, kwargs: dict):
+        """
+        Validate types of user-provided "named aggregation" kwargs.
+
+        `TypeError` is raised if aggfunc is not `str` or callable.
```
Comment (YarShev) [inline modin/pandas/groupby.py:1931]: This should go under `Raises` section.

File: modin/pandas/groupby.py [hunk 3]
```diff
@@ -1924,8 +1924,45 @@ def nsmallest(self, n=5, keep="first"):
...
+        Returns
+        -------
+        columns : List[str]
+            List of user-provided keys.
+        func : List[Union[str, callable[...,Any]]]
+            List of user-provided aggfuncs.
+
+        Notes
+        -----
+        Copied from pandas
```
Comment (YarShev) [inline modin/pandas/groupby.py:1946]:
```suggestion
        Copied from pandas.
```

File: modin/pandas/groupby.py [hunk 4]
```diff
@@ -1924,8 +1924,45 @@ def nsmallest(self, n=5, keep="first"):
...
+        """
+        tuple_given_message = "func is expected but received {} in **kwargs."
+        columns = list(kwargs)
+        func = []
```
Comment (YarShev) [inline modin/pandas/groupby.py:1950]:
```suggestion
        funcs = []
```

File: modin/pandas/test/test_groupby.py [hunk 5]
```diff
@@ -2989,6 +2989,14 @@ def test_groupby_apply_series_result(modify_config):
     )
 
 
+def test_groupby_named_aggregation():
+    ser = pd.Series([10, 10, 10, 1, 1, 1, 2, 3], name="data")
+
+    eval_general(
+        ser, ser._to_pandas(), lambda ser: ser.groupby(level=0).agg(result=("max"))
```
Comment (YarShev) [inline modin/pandas/test/test_groupby.py:2996]: I think we should directly create a pandas Series but not use _to_pandas.

**Comment by anmyachev** [comment]
@YarShev all your comments are addressed.

**Review by YarShev — DISMISSED** [review]

**Review by dchigarev — APPROVED** [review]
LGTM

**Pull request merged**